src/core/pyproxy.c

```diff
@@ -618,6 +618,105 @@ _pyproxy_ensure_future(PyObject* pyobject,
   return success ? 0 : -1;
 }
 
+///////////////////////////////////////////////////////////////////////////////
+//
+// Buffers
+//
+
+// For debug
+size_t py_buffer_len_offset = offsetof(Py_buffer, len);
+size_t py_buffer_shape_offset = offsetof(Py_buffer, shape);
+
+bool
+_pyproxy_is_buffer(PyObject* ptrobj)
+{
+  return PyObject_CheckBuffer(ptrobj);
+}
+
+JsRef
+array_to_js(Py_ssize_t* array, int len)
+{
+  JsRef result = hiwire_array();
+  for (int i = 0; i < len; i++) {
+    JsRef val = hiwire_int(array[i]);
+    hiwire_push_array(result, val);
+    hiwire_decref(val);
+  }
+  return result;
+}
+
+// The order of these fields has to match the code in getRawBuffer
+typedef struct
+{
+  void* start_ptr;
+  void* smallest_ptr;
+  void* largest_ptr;
+  JsRef shape;
+  JsRef strides;
+  Py_buffer* view;
+} buffer_struct;
+
+buffer_struct*
+_pyproxy_memoryview_get_buffer(PyObject* ptrobj)
+{
+  if (!PyObject_CheckBuffer(ptrobj)) {
+    return NULL;
+  }
+  Py_buffer view;
+  if (PyObject_GetBuffer(ptrobj, &view, PyBUF_RECORDS_RO) == -1) {
+    PyErr_Clear();
+    return NULL;
+  }
+
+  bool success = false;
+  if (view.ndim == 0) {
+    FAIL();
+  }
+  buffer_struct result = { 0 };
+  result.shape = array_to_js(view.shape, view.ndim);
+
+  result.start_ptr = result.smallest_ptr = result.largest_ptr = view.buf;
```

Comment:
This could use some comments particularly about what what `smallest_ptr`, `largest_ptr` etc do